src/lib.rs

```diff
@@ -88,6 +90,22 @@ impl std::fmt::Display for Build {
     }
 }
 
+impl codicon::Decoder<()> for Build {
+    type Error = std::io::Error;
+
+    fn decode(mut reader: impl Read, _: ()) -> std::io::Result<Self> {
```

Comment:
Interesting function signature - does this just mean there are no parameters?

Reply:
Correct, the semantics for not passing a parameter type is just to pass a unit struct.

Alternatively I could have done `impl codicon::Decoder<Blah> for Build` if the `Blah` struct had state that I'd find useful for this

Reply:
That is cool, hadn't seen this before.